src/main/java/teammates/storage/entity/FeedbackQuestion.java

```diff
@@ -119,8 +119,14 @@ public void setLastUpdate(Instant newDate) {
         this.updatedAt = newDate;
     }
 
+    /**
+     * Gets the question id, in legacy key format similar to that generated by the old appengine SDK.
+     */
     public String getId() {
-        return Key.create(FeedbackQuestion.class, feedbackQuestionId).toLegacyUrlSafe();
+        String untrimmedId = Key.create(FeedbackQuestion.class, feedbackQuestionId).toLegacyUrlSafe();
+
+        // Trim last 4 characters for backward compatibility of keys that were generated before Objectify V6
+        return untrimmedId.substring(0, untrimmedId.length() - 4);
```

Comment:
I would say to provide more information/background here on how we reach this solution?

Reply:
Right! Do you think here we should make other devs aware that the region (if they were to deploy to staging) can affect the backward compatibility of the key? Or shall we include that as a warning in documentation, I think it's pretty important as it can easily cause confusion.

Reply:
No need to waste effort on that. The only reason for us testing cross-compatibility is for the actual user's data. We don't realistically need developers' test data to have the same compatibility (test data are very disposable and easily generated).